scripts/uosc/elements/Menu.lua

```diff
@@ -212,6 +215,29 @@ function Menu:update(data)
 
 	self:update_content_dimensions()
 	self:reset_navigation()
+
+	-- Ensure palette menus have active searches, and clean empty searches from menus that lost the `palette` flag
+	local update_dimensions_again = false
+	for _, menu in ipairs(self.all) do
+		if menu.palette and not menu.search then
+			update_dimensions_again = true
+			self:search_init(menu)
+		elseif not menu.palette and menu.search and menu.search.query == '' then
+			update_dimensions_again = true
+			menu.search = nil
+		end
+	end
+	-- We update before _and_ after because search_inits need the initial un-searched
+	-- menu's position and scroll state to save on the `search.source` table.
+	if update_dimensions_again then
+		self:update_content_dimensions()
+		self:reset_navigation()
+	end
+	for _, menu in ipairs(new_menus) do
+		print(menu.id..': '.. (menu.initial_query or ''))
+		if menu.initial_query then self:search_query_update(menu.initial_query, menu) end
```

Comment:
I think that crashes if there isn't a currently active search (not a palette).

`search_query_update()` will cause a search to be executed, which is probably not what we want here?

And a `print()` slipped in.